### PR TITLE
CB-20104 Handle AWS InsufficientInstanceCapacityException during stop…

### PR DESCRIPTION
**Pull request by aakashshankar**
…start upscale

See detailed description in the commit message.

### CODE REVIEW

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java
Comment (lajosrodek) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java:1]: Please cover the new logic with UT.

File: cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java
Comment (lajosrodek) [inline cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java:1]: Please cover the new logic with UT.

File: cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java [hunk 1]
```diff
@@ -124,22 +129,35 @@ private List<CloudVmInstanceStatus> startInstances(
             // We exceeded the time-bound on the polling. Try getting the status for all the instances that we attempted to start.
             LOGGER.warn("Timed out while attempting to start instances. Attempting to get states for attempted nodes", p);
             return getInstanceStatusOnStartError(connector, ac, instancesToStart, p);
+        } catch (InsufficientCapacityException ice) {
+            LOGGER.error("Insufficient capacity on the cloud provider while attempting to start instances. Trying to fetch statuses of nodes");
```
Comment (lajosrodek) [inline cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java:133]: Please also log `ice`.

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java
Comment (aakashshankar) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java:1]: Added UT

File: cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java (hunk 1 above)
Comment (aakashshankar) [reply cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java]: Logged `ice`

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

**Pull request merged**